### ISSUE TITLE
Replace Date.today with Date.current in the rspec specs

### ISSUE THREAD

**Issue by AEtherC0r3** [openSUSE/osem#1522]
The test in https://github.com/openSUSE/osem/blob/master/spec/models/program_spec.rb#L154-L157 is passing because of a false negative. More specifically, in line 155 it uses build instead of create, so the new cfp isn't saved in the db, therefore Program#cfp_open? returns false because cfp.psesent? is false in https://github.com/openSUSE/osem/blob/master/app/models/program.rb#L140 and not because the cfp is over. If create is used instead of build then it will fail (if Date.today > Date.current). Maybe Date.current should be used instead of Date.today in general, since Date.today takes into account the server's local timezone.

**Comment by Ana06** [comment]
@AEtherC0r3 good catch! the cfp should definitely be saved in thr test and `Date.current` should be used, as it is what is also used in the `cfp_open` method. 

@AEtherC0r3 will send a PR? :wink:

There is also a nice RoboCop cop to ensure that the timezone it is always used: https://rubocop.readthedocs.io/en/latest/cops_rails/#railstimezone. It would also be nice to enable it with the *flexible* option. I think there would be some offenses tricky to solve, so this should be made in a different PR. @AEtherC0r3 would you also like to do this or should I create a new issue?

**Comment by AEtherC0r3** [comment]
@Ana06 I will do both of them (the fix and the cop)

**Comment by Ana06** [comment]
@AEtherC0r3 great! 😉

**Comment by greggawatt** [comment]
is this issued fixed?

**Comment by Ana06** [comment]
No, this part is not done:

> There is also a nice RoboCop cop to ensure that the timezone it is always used: https://rubocop.readthedocs.io/en/latest/cops_rails/#railstimezone. It would also be nice to enable it with the flexible option. I think there would be some offenses tricky to solve, so this should be made in a different PR. @AEtherC0r3 would you also like to do this or should I create a new issue?

@AEtherC0r3 do you still plan to work on this? :thinking:

**Comment by AEtherC0r3** [comment]
I still want to work on that but I don't think that is going to happen soon. So, @greggawatt if you want to do it, then go ahead :)

**Comment by kkavita92** [comment]
@Ana06 happy to jump on this if it's still open! Let me know :)

**Comment by differentreality** [comment]
@kkavita92 you can go ahead and enable the cop!

**Comment by Anthematics** [comment]
I could maybe take a look at it -> I'm a jr. developer who is about to finish a boot camp -. my stack is ruby (and rails), JS, and I can work with JSON  & My SQL. 

I have never contributed to open source but would love to try!

**Comment by differentreality** [comment]
Hi @Anthematics, welcome :)

@kkavita92 wanted to work on this some time ago. @kkavita92 please let us know if you are working on this, otherwise someone else could take it up.

@Anthematics there are also other issues to choose from, if you want to. You can get started with our [junior issues](https://github.com/openSUSE/osem/labels/Junior)!
